backend/testsuite/io_integration_test.go

```diff
@@ -0,0 +1,557 @@
+package testsuite
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"regexp"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+
+	"github.com/c2fo/vfs/v6"
+	"github.com/c2fo/vfs/v6/options"
+	"github.com/c2fo/vfs/v6/vfssimple"
+)
+
+type OSWrapper struct {
```

Comment:
OSWrapper implements ReadWriteSeekCloseURINamer used by this test to generically test io functionality for vfs.File implemenations.  However, I needed a baseline to model the tests on so this basically provides an unix os file interface (outside of the os backend) to ensure we're following those io behaviors.